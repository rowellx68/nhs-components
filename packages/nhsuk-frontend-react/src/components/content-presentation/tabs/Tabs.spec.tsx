import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { composeStory } from '@storybook/react';
import meta, { Default as DefaultStory } from './Tabs.stories';

const Default = composeStory(DefaultStory, meta);

it('should render the Tabs component', async () => {
  const { container } = render(<Default />);

  expect(container).toMatchSnapshot();

  const secondTab = container.querySelector('[href="#past-week"]');

  await userEvent.click(secondTab!);

  expect(
    container.querySelector('[href="#past-week"]')?.parentElement,
  ).toHaveClass('nhsuk-tabs__list-item--selected');
});
