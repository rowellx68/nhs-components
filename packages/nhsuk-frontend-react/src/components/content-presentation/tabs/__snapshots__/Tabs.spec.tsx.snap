// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the Tabs component 1`] = `
<div>
  <div
    class="nhsuk-tabs"
    data-module="nhsuk-tabs"
  >
    <h2
      class="nhsuk-tabs__title"
    >
      Contents
    </h2>
    <ul
      class="nhsuk-tabs__list"
      role="tablist"
    >
      <li
        class="nhsuk-tabs__list-item nhsuk-tabs__list-item--selected"
        role="presentation"
      >
        <a
          aria-controls="past-day"
          aria-selected="true"
          class="nhsuk-tabs__tab"
          href="#past-day"
          id="tab_past-day"
          role="tab"
          tabindex="0"
        >
          Past day
        </a>
      </li>
      <li
        class="nhsuk-tabs__list-item"
        role="presentation"
      >
        <a
          aria-controls="past-week"
          aria-selected="false"
          class="nhsuk-tabs__tab"
          href="#past-week"
          id="tab_past-week"
          role="tab"
          tabindex="-1"
        >
          Past week
        </a>
      </li>
      <li
        class="nhsuk-tabs__list-item"
        role="presentation"
      >
        <a
          aria-controls="past-month"
          aria-selected="false"
          class="nhsuk-tabs__tab"
          href="#past-month"
          id="tab_past-month"
          role="tab"
          tabindex="-1"
        >
          Past month
        </a>
      </li>
      <li
        class="nhsuk-tabs__list-item"
        role="presentation"
      >
        <a
          aria-controls="past-year"
          aria-selected="false"
          class="nhsuk-tabs__tab"
          href="#past-year"
          id="tab_past-year"
          role="tab"
          tabindex="-1"
        >
          Past year
        </a>
      </li>
    </ul>
    <div
      aria-labelledby="tab_past-day"
      class="nhsuk-tabs__panel"
      id="past-day"
      role="tabpanel"
    >
      <h2>
        Past day
      </h2>
      <p>
        Content for the past day
      </p>
    </div>
    <div
      aria-labelledby="tab_past-week"
      class="nhsuk-tabs__panel nhsuk-tabs__panel--hidden"
      id="past-week"
      role="tabpanel"
    >
      <h2>
        Past week
      </h2>
      <p>
        Content for the past week
      </p>
    </div>
    <div
      aria-labelledby="tab_past-month"
      class="nhsuk-tabs__panel nhsuk-tabs__panel--hidden"
      id="past-month"
      role="tabpanel"
    >
      <h2>
        Past month
      </h2>
      <p>
        Content for the past month
      </p>
    </div>
    <div
      aria-labelledby="tab_past-year"
      class="nhsuk-tabs__panel nhsuk-tabs__panel--hidden"
      id="past-year"
      role="tabpanel"
    >
      <h2>
        Past year
      </h2>
      <p>
        Content for the past year
      </p>
    </div>
  </div>
</div>
`;
